internal/services/eventhub/eventhub_cluster_resource.go

```diff
@@ -153,7 +153,7 @@ func resourceEventHubClusterDelete(d *pluginsdk.ResourceData, meta interface{})
 	return pluginsdk.Retry(d.Timeout(pluginsdk.TimeoutDelete), func() *pluginsdk.RetryError {
 		future, err := client.ClustersDelete(ctx, *id)
 		if err != nil {
-			if strings.Contains(err.Error(), "Cluster cannot be deleted until four hours after its creation time") || response.WasStatusCode(future.HttpResponse, http.StatusTooManyRequests) {
+			if strings.Contains(err.Error(), "cannot be deleted until at least 4 hours after its created time") || response.WasStatusCode(future.HttpResponse, http.StatusTooManyRequests) {
```

Comment:
string matching is unreliable, can we match on the Code instead of the Message, which shouldn't change?

Reply:
Thanks Tom for the comment, I agree with you about the fact that matching the error message would be unreliable as it possibly to be changed by API at any time.

I check the HTTP code of this deletion request is 400, which is a very general bad request code, do you think we also need to match the error message along with the code to improve the matching accuracy?

Reply:
Thanks Tom, I added the status code check but kept the error message matching as well, let me know about your concerns if there is any. :)